web/src/TabNav.tsx

```diff
@@ -139,56 +143,83 @@ export function OverviewNavProvider(
       return
     }
 
-    setTabs((prevState) => {
-      if (prevState.includes(name)) {
-        return prevState
-      }
+    if (tabs.includes(name) && name == selectedTab) {
```

Comment:
my understanding is that it's safer to call `setState` with `(prevState: S) => S` rather than just `S` (and this is what the old code does)

The new code is using the closed value of `tabs` from the start of the function, which might always have the value we want, but I'm not certain, so I've tended to err on the side of updating the previous value at the time of the callback invocation.

Also, this: https://medium.com/ableneo/react-setstate-does-not-immediately-update-the-state-84dbd26f67d5

Maybe this is all fine as-is and will always do what we want. Maybe at some point we end up with an annoying-to-diagnose bug as a result from it, or maybe that's impossible. I haven't really spent time trying to convince myself it's safe to rely on the closed value and would be happy to learn it's fine.

Reply:
we can also just leave this as-is and trust that if it is actually a problem, bug reports will be soon enough that we'll remember this as a potential issue

Reply:
(left a comment about this)